File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False

        word1_chars_map = Counter(word1)
        word2_chars_map = Counter(word2)

        if len(word1_chars_map) != len(word2_chars_map):
            return False

        arr1 = []
        arr2 = []

        for key, value in word1_chars_map.items():
            if key not in word2_chars_map:
                return False
            
            arr1.append(value)
            arr2.append(word2_chars_map[key])
        
        arr1.sort()
        arr2.sort()

        for i in range(len(arr1)):
            if arr1[i] != arr2[i]:
                return False
        
        return True

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        // Meta Prep Time Solution 1:
        int[] freq1 = new int[26];
        int[] freq2 = new int[26];

        for(char curr : word1.toCharArray()){
            freq1[curr - 'a'] += 1;
        }

        for(char curr : word2.toCharArray()){
            freq2[curr - 'a'] += 1;
        }

        // Check if character is present or not?
        for(int i = 0; i < 26; i++){
            boolean isW1CharPresent = freq1[i] == 0;
            boolean isW2CharPresent = freq2[i] == 0;

            if(isW1CharPresent != isW2CharPresent){
                return false;
            }
        }

        // Check equal frequencies;
        Arrays.sort(freq1);
        Arrays.sort(freq2);

        for(int i = 0; i < 26; i++){
            if(freq1[i] != freq2[i]){
                return false;
            }
        }

        return true;
    }
}